core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -917,24 +973,53 @@ private void endRequest0(@Nullable Throwable requestCause, long requestEndTimeNa
             }
         }
 
+        if (!needToDeferName()) {
+            setNamesIfAbsent();
+        }
+        this.requestEndTimeNanos = requestEndTimeNanos;
+        this.requestCause = requestCause;
+        updateFlags(flags);
+    }
+
+    private void setNamesIfAbsent() {
         if (name == null) {
+            String newServiceName = null;
             String newName = null;
-            final RpcRequest rpcReq = ctx.rpcRequest();
+            RpcRequest rpcReq = ctx.rpcRequest();
+            if (rpcReq == null && requestContent instanceof RpcRequest) {
+                rpcReq = (RpcRequest) requestContent;
+            }
+
             if (rpcReq != null) {
+                newServiceName = rpcReq.serviceType().getName();
                 newName = rpcReq.method();
-            } else if (requestContent instanceof RpcRequest) {
-                newName = ((RpcRequest) requestContent).method();
+            } else if (ctx instanceof ServiceRequestContext) {
+                final ServiceConfig config = ((ServiceRequestContext) ctx).config();
+                newServiceName = config.defaultServiceName();
+                if (newServiceName == null) {
+                    newServiceName = config.service().unwrap().getClass().getName();
```

Comment:
So, we're not using a simple name, right?

Reply:
Yes, the `serviceName` of gRPC is FQCN. So I thought we use FQCN for `serviceName` except for access logs.

Reply:
Cool. Thanks.